engine/src/test/java/org/camunda/bpm/engine/test/bpmn/el/ExpressionManagerTest.java

```diff
@@ -138,7 +151,38 @@ public void testJuelExpressionWithNonPublicClass() {
         .activityId("userTask")
         .singleResult();
     assertThat(userTask).isNotNull();
+  }
+
+  @Test
+  @RequiredHistoryLevel(ProcessEngineConfiguration.HISTORY_FULL)
+  public void shouldCompareWithBigDecimal() {
+    // given
+    BpmnModelInstance process = Bpmn.createExecutableProcess("testProcess")
+        .startEvent()
+          .exclusiveGateway()
+            .condition("true", "${total.compareTo(myValue) >= 0}")
+            .userTask("userTask")
```

Comment:
Do we need an end event here? I think it might be implicit but not sure.

Reply:
Every task without outgoing sequence flow is a "sink", i.e., the task consumes the token. The only reason to add the end event would be to follow a best practice. Technically, it works even without the end event.

Reply:
I was thinking something like this. I am fine with either approach. Feel free to merge. :)